// ddr-configuration/fesm2015/ddr-configuration.js
import { Injectable, ɵɵdefineInjectable, ɵɵinject, Pipe, NgModule } from '@angular/core';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { has, get, cloneDeep } from 'lodash';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DdrConfigurationService {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
    }
    /**
     * @param {?} pathJSON
     * @return {?}
     */
    getDataFromJSON(pathJSON) {
        return new Promise((/**
         * @param {?} resolve
         * @param {?} reject
         * @return {?}
         */
        (resolve, reject) => {
            this.http.get(pathJSON).subscribe((/**
             * @param {?} data
             * @return {?}
             */
            data => {
                this._data = data;
                resolve(true);
            }), (/**
             * @param {?} error
             * @return {?}
             */
            error => {
                console.error("Ddr-Conf: ", error);
                reject(true);
            }));
        }));
    }
    /**
     * @param {?} path
     * @return {?}
     */
    getData(path) {
        if (!has(this._data, path)) {
            console.error("Not exists path: " + path);
            return null;
        }
        else {
            return get(this._data, path);
        }
    }
    /**
     * @return {?}
     */
    getAllData() {
        return cloneDeep(this._data);
    }
}
DdrConfigurationService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
DdrConfigurationService.ctorParameters = () => [
    { type: HttpClient }
];
/** @nocollapse */ DdrConfigurationService.ngInjectableDef = ɵɵdefineInjectable({ factory: function DdrConfigurationService_Factory() { return new DdrConfigurationService(ɵɵinject(HttpClient)); }, token: DdrConfigurationService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    DdrConfigurationService.prototype._data;
    /**
     * @type {?}
     * @private
     */
    DdrConfigurationService.prototype.http;
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class GetDataConfPipe {
    /**
     * @param {?} ddrConfiguration
     */
    constructor(ddrConfiguration) {
        this.ddrConfiguration = ddrConfiguration;
    }
    /**
     * @param {?} path
     * @return {?}
     */
    transform(path) {
        return this.ddrConfiguration.getData(path);
    }
}
GetDataConfPipe.decorators = [
    { type: Pipe, args: [{
                name: 'getDataConf'
            },] }
];
/** @nocollapse */
GetDataConfPipe.ctorParameters = () => [
    { type: DdrConfigurationService }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    GetDataConfPipe.prototype.ddrConfiguration;
}

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DdrConfigurationModule {
}
DdrConfigurationModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    GetDataConfPipe
                ],
                imports: [
                    HttpClientModule
                ],
                exports: [GetDataConfPipe]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { DdrConfigurationModule, DdrConfigurationService, GetDataConfPipe as ɵa };
