// ddr-configuration/esm5/lib/ddr-configuration.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import * as _ from 'lodash';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
var DdrConfigurationService = /** @class */ (function () {
    function DdrConfigurationService(http) {
        this.http = http;
    }
    /**
     * @param {?} pathJSON
     * @return {?}
     */
    DdrConfigurationService.prototype.getDataFromJSON = /**
     * @param {?} pathJSON
     * @return {?}
     */
    function (pathJSON) {
        var _this = this;
        return new Promise((/**
         * @param {?} resolve
         * @param {?} reject
         * @return {?}
         */
        function (resolve, reject) {
            _this.http.get(pathJSON).subscribe((/**
             * @param {?} data
             * @return {?}
             */
            function (data) {
                _this._data = data;
                resolve(true);
            }), (/**
             * @param {?} error
             * @return {?}
             */
            function (error) {
                console.error("Ddr-Conf: ", error);
                reject(true);
            }));
        }));
    };
    /**
     * @param {?} path
     * @return {?}
     */
    DdrConfigurationService.prototype.getData = /**
     * @param {?} path
     * @return {?}
     */
    function (path) {
        if (!_.has(this._data, path)) {
            console.error("Not exists path: " + path);
            return null;
        }
        else {
            return _.get(this._data, path);
        }
    };
    /**
     * @return {?}
     */
    DdrConfigurationService.prototype.getAllData = /**
     * @return {?}
     */
    function () {
        return _.cloneDeep(this._data);
    };
    DdrConfigurationService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    DdrConfigurationService.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    /** @nocollapse */ DdrConfigurationService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function DdrConfigurationService_Factory() { return new DdrConfigurationService(i0.ɵɵinject(i1.HttpClient)); }, token: DdrConfigurationService, providedIn: "root" });
    return DdrConfigurationService;
}());
export { DdrConfigurationService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    DdrConfigurationService.prototype._data;
    /**
     * @type {?}
     * @private
     */
    DdrConfigurationService.prototype.http;
}
